model-optimizer/mo/back/offline_transformations.py

```diff
@@ -3,28 +3,48 @@
 
 import argparse
 
-if __name__ == "__main__":
-    parser = argparse.ArgumentParser()
-    parser.add_argument("--input_model")
-    parser.add_argument("--framework")
-    args = parser.parse_args()
-    path_to_model = args.input_model
+from mo.utils.error import Error
+from mo.utils.cli_parser import parse_transform
 
-    # This variable is only needed by GenerateMappingFile transformation
-    # to produce correct mapping
-    extract_names = True if args.framework in ['tf', 'mxnet', 'kaldi'] else False
 
+def get_available_transformations():
     try:
-        from openvino.inference_engine import IECore, read_network # pylint: disable=import-error
-        from openvino.offline_transformations import ApplyMOCTransformations, GenerateMappingFile, CheckAPI # pylint: disable=import-error
+        from openvino.offline_transformations import ApplyLowLatencyTransformation # pylint: disable=import-error
+        return {
+            'LowLatency': ApplyLowLatencyTransformation,
+        }
     except Exception as e:
-        print("[ WARNING ] {}".format(e))
-        exit(1)
+        return {}
+
+
+def apply_offline_transformations(input_model: str, framework: str, transforms: list):
+    # This variable is only needed by GenerateMappingFile transformation
+    # to produce correct mapping
+    extract_names = True if framework in ['tf', 'mxnet', 'kaldi'] else False
 
-    CheckAPI()
+    from openvino.inference_engine import read_network # pylint: disable=import-error
+    from openvino.offline_transformations import ApplyMOCTransformations, GenerateMappingFile # pylint: disable=import-error
 
-    net = read_network(path_to_model + "_tmp.xml", path_to_model + "_tmp.bin")
-    net.serialize(path_to_model + ".xml", path_to_model + ".bin")
-    path_to_mapping = path_to_model + ".mapping"
+    net = read_network(input_model + "_tmp.xml", input_model + "_tmp.bin")
```

Comment:
May be introduce function to generate temporary IR xml and bin files names so you can use them in place where you create the temporary model and here. This will allow you to not hardcode `_tmp` suffixes in two different files

Reply:
Actually we have this prepare_and_emmit_ir function which generates an IR but the problem is that we execute offline transformations in sub process to catch runtime errors. 
I agree that this needs to be simplified but I would postpone to the next refactoring PR.